### PR TITLE
Login and Logout Page

### PR DESCRIPTION
**Pull request by samnangcattor**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in(user)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<% provide(:title, "Log in") %>
```
Comment (dieunb) [inline app/views/sessions/new.html.erb]: remove `()`

File: app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+<h1>Log in</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(:session, url: login_path) do |f| %>
```
Comment (dieunb) [inline app/views/sessions/new.html.erb]: remove `()`

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -12,6 +12,9 @@
   <body>
     <%= render 'layouts/header' %>
```
Comment (RathanakSreang) [inline app/views/layouts/application.html.erb]: Use " instead of '

File: app/views/sessions/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<% provide :title, "Log in" %>
+<h1>Log in</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for :session, url: login_path do |f| %>
+
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
```
Comment (RathanakSreang) [inline app/views/sessions/new.html.erb]: User " instead of '

File: app/views/sessions/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+      <%= f.label :password %>
+      <%= f.password_field :password, class: 'form-control' %>
```
Comment (RathanakSreang) [inline app/views/sessions/new.html.erb]: User " instead of '

File: config/routes.rb [hunk 10]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
-
   root             'static_pages#home'
+  get    'login'   => 'sessions#new'
+  post   'login'   => 'sessions#create'
+  delete 'logout'  => 'sessions#destroy'
```
Comment (RathanakSreang) [inline config/routes.rb]: User " instead of '

File: app/controllers/sessions_controller.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to root_url
+    else
+      flash[:danger] = 'Invalid email/password combination'
```
Comment (RathanakSreang) [inline app/controllers/sessions_controller.rb]: User " instead of '

File: Gemfile [hunk 12]
```diff
@@ -3,6 +3,7 @@ source 'https://rubygems.org'
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '4.2.0'
+gem 'bcrypt',               '3.1.7'
 gem 'bootstrap-sass',       '3.2.0.0'
 # Use mysql as the database for Active Record
 gem 'mysql2'
```
Comment (RathanakSreang) [inline Gemfile]: User " instead of '

File: db/seeds.rb [hunk 13]
```diff
@@ -5,3 +5,5 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+
+User.create(name: 'a', email: 'a@gmail.com', password: '123456', password_confirmation: '123456')
```
Comment (RathanakSreang) [inline db/seeds.rb]: User " instead of ' and remove ()

File: app/controllers/sessions_controller.rb [hunk 14]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
```
Comment (HorNorin) [inline app/controllers/sessions_controller.rb]: remove `()` from authenticate

File: db/seeds.rb [hunk 15]
```diff
@@ -3,5 +3,7 @@
 #
 # Examples:
 #
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+#   cities = City.create([{ name: "Chicago" }, { name: "Copenhagen" }])
+#   Mayor.create(name: "Emanuel", city: cities.first)
+
+User.create(name: "a", email: "a@gmail.com", password: "123456", password_confirmation: "123456")
```
Comment (HorNorin) [inline db/seeds.rb]: remove `()`

File: app/controllers/sessions_controller.rb (hunk 14 above)
Comment (samnangcattor) [inline app/controllers/sessions_controller.rb]: remove authenticate is  error

**Pull request merged**